### ISSUE TITLE
List of software that doesn't work

### ISSUE THREAD

**Issue by simontoens** [emufreak/iAmiga#31]
This issue exists to track the list of software that does not work with iAmiga.  We can create more detailed issues once we figure out the actual cause.

Defender of the Crown (runs too fast)*
It came from the Desert (runs too fast)*
Wings (runs too fast)*
Jumping Jackson (runs too fast)*
Firepower (runs too fast)*
IK+ (no sound)

\* possibly related to the processor being emulated?  I am not even sure which particular Amiga model we are emulating - being able to adjust the processor type may be an issue in itself.

Xenon 2 (bad graphics sometimes - race condition?)
The Immortal (bad graphics)
Rainbow Islands (does not load)

**Comment by mithrendal** [comment]
"Xenon2 -Megablast ... [cr VF].adf" the one disk only version is perfectly playable with good graphics. Distributions with two disks do crash though. And yes the game is a blast ;-)

(It is also mentioned in http://arstechnica.com/gadgets/2015/07/the-amiga-turns-30-nobody-had-ever-designed-a-personal-computer-this-way/   at Part 7 -> Page 3)

**Comment by simontoens** [comment]
Thanks @mithrendal, good to know, I will test.

**Comment by simontoens** [comment]
@mithrendal I still see the same problem, let's talk offline - can you email me at simontoens@gmail.com?

**Comment by simontoens** [comment]
Turns out that there some sort of race condition with Xenon 2 - sometimes the graphics look ok, often they don't.

**Comment by emufreak** [comment]
I've given the issues with speed some thought. Browsing the web I found two main reasons for this. Cpu speed and Vsync emulation. VSync is used as a sort of time measurement for some games. As I understand it all this components run as fast as possible at the moment. 

@simontoens maybe you can test if this is fixed in uae4all2 on some of this games.

**Comment by simontoens** [comment]
@emufreak yes I can test with uae4all2 - this will take a while because I just got a new machine and I will have to re-setup uae4all2.

**Comment by emufreak** [comment]
@simontoens Thanks

**Comment by emufreak** [comment]
Just found out that Defender of the Crown runs fine for me if i switch on NTSC settings

**Comment by simontoens** [comment]
hmmm I don't see any difference.  Which adf version are you using?

**Comment by mithrendal** [comment]
@simontoens this week I have got Xenon2 working now on my iPhone... 🤤 


![FF5A92B1-ADA9-4208-9082-9A6A5E04F0A5](https://user-images.githubusercontent.com/17108995/137538796-6420816b-acb5-4454-9733-608079448646.png)

took a long journey to come to this result but 😍 I like it ... 

the culprit here is exact emulation timing 🙄 as @emufreak already said....

... my dream some years ago was to migrate iAmiga to another emulation core... looking at the current WinUAE and implement it in iAmiga was a rather overwhelming task to be done for me ... the source code was to complex and confusing ...

then there were the very efficient and clean coded emulation cores from @dirkwhoffmann ...

I first tried to play with a simpler machine the C64 and did https://vc64web.github.io I also integrated  csdb.dk (scene database) gallery browser 🤤 which is really cool I think to browse all the demos out there... 

I brought over some concepts of iAmiga e.g. the @simontoens custom buttons and snapshot galery ...

And in the last week I did the same with the Amiga 🤤  I call it vAmigaWeb ... it is deployed at https://vAmigaWeb.github.io ... it still lacks keyboard and mouse input ... but joysticks are working good ... missing parts will come in the next couple of days ... and I still need a demo site to get stuff from ...

both webapps follows the PWA standard which means you can save it to iPhones homescreen and then it offline installs and behaves nearly like a native app. When it is installed it will be launch completely from local iPhone no network traffic needed anymore unless you upgrade the app of course ... 

the github repo with the source code and the emulation core is at https://github.com/vAmigaWeb/vAmigaWeb

so yeah ... I think iAmiga project should wipe the ancient not very exact core and replace it with the same very compact and clean vAmiga core and then all the incompatibilities will be no more ...😍

**Comment by simontoens** [comment]
@mithrendal great to hear from you and also this looks amazing.  I will try it out right away.  I am actually working on another emulator attempt of the best computer ever (sorry I can't tell if you are leaning more towards C64 or Amiga - you know you will have to decide at some point 😈 ) - I was (well I am) trying to port fs-uae to iOS.  But that isn't trivial because of dependency hell. I am going to try some of the programs I could never get working on iAmiga!

**Comment by mithrendal** [comment]
>... trying to port fs-uae to iOS.

well FSUAE is also a hell to understand .... I thought of it too to bring it in iAmiga in the past ... I would say after my experiences with vAmiga ... that the update to the vAmiga core is by far ☝️!!! ... more simple ...

advantages:
- it is already a XCode project ... 
- it has practically no dependencies ...
- it is a complete new and clean code base ... no historic relicts ... 
- just drop the Emulator folder into iAmiga, just new Amiga(); run it and then grab the screen buffer ... 
- it is highly compatible 😍... @dirkwhoffmann has written a test suite https://github.com/dirkwhoffmann/vAmigaTS for it ...which can be run on real amigas to compare the results of vAmigaEmulator. We had some sessions with Toni Wilen 🙏🏻 who helps the project in difficult questions with his outstanding knowledge ...
- it is actively developed ... development started in 2019 from the scratch


>I am going to try some of the programs I could never get working on iAmiga!

yeah good idea 🤓 before you decide to use the vAmiga core in iAmiga ... you could test vAmigas emulation core on your favourites  
  - on Mac with https://dirkwhoffmann.github.io/vAmiga/
  - on iOS with https://vamigaweb.github.io/   (save that puppy to homescreen to get rid of the safari web gestures and safari address bar... installed to homescreen it will feel like a native iOS app)

**Comment by simontoens** [comment]
Definitely the "no dependencies" thing is a huge advantage...
